### PR TITLE
[WIP] E-Branchformer Encoder in ESPnet2

### PR DESCRIPTION
**Pull request by kkim-asapp**
Hi, 
This PR is about the E-Branchformer encoder (Kim et al., SLT 2022: will be appeared, Arxiv: [link](https://arxiv.org/abs/2210.00077)). 

The module is implemented in ESPnet2 with the base recipe for Librispeech ASR.
Also, I've adding an argument in `Repeat` module to support LayerDrop ([link](https://arxiv.org/abs/1909.11556)). 

This PR is in progress, and I will update results on README file.

### CODE REVIEW

File: espnet2/asr/encoder/e_branchformer_encoder.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+        )
+        self.merge_proj = torch.nn.Linear(size + size, size)
+
+    def forward(self, x_input, mask, cache=None):
+        """Compute encoded features.
+        Args:
+            x_input (Union[Tuple, torch.Tensor]): Input tensor w/ or w/o pos emb.
+                - w/ pos emb: Tuple of tensors [(#batch, time, size), (1, time, size)].
+                - w/o pos emb: Tensor (#batch, time, size).
+            mask (torch.Tensor): Mask tensor for the input (#batch, time).
```
Comment (pyf98) [inline espnet2/asr/encoder/e_branchformer_encoder.py:109]: The shape of `mask` is `(#batch, 1, time)`?

**Comment by pyf98** [comment]
Looks good to me!

Since the `repeat` function is updated to support the new Layer Drop, we may need to update the other encoders as well (probably in a later PR?).

**Comment by sw005320** [comment]
@kkim-asapp, thanks a lot for the great PR!
Can you fix the format error? 
https://github.com/espnet/espnet/actions/runs/3632971395/jobs/6129594548#step:8:12

**Comment by kkim-asapp** [comment]
> @kkim-asapp, thanks a lot for the great PR! Can you fix the format error? https://github.com/espnet/espnet/actions/runs/3632971395/jobs/6129594548#step:8:12

I did!

File: espnet2/asr/encoder/e_branchformer_encoder.py (hunk 1 above)
Comment (pyf98) [reply espnet2/asr/encoder/e_branchformer_encoder.py:109]: Hi @kkim-asapp Is the shape correct?

Comment (kkim-asapp) [reply espnet2/asr/encoder/e_branchformer_encoder.py:109]: Thanks, @pyf98! You're right. It was wrong. Let me update it.

Comment (pyf98) [reply espnet2/asr/encoder/e_branchformer_encoder.py:109]: I remember it was wrong in Conformer. If so, can you update it as well?

Comment (kkim-asapp) [reply espnet2/asr/encoder/e_branchformer_encoder.py:109]: Sure, will do.

**Comment by sw005320** [comment]
You can run some of the test scripts locally instead of checking it via CI.
https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md#4-unit-testing

**Comment by kkim-asapp** [comment]
> You can run some of the test scripts locally instead of checking it via CI. https://github.com/espnet/espnet/blob/master/CONTRIBUTING.md#4-unit-testing

I did, but the error logs are different. Let me check after reinstall dependencies.

**Comment by sw005320** [comment]
Oh, OK.
It also depends on the version of the test script and it would not be the exactly same.

**Comment by kkim-asapp** [comment]
Hi @pyf98, is there anything I can do to fix one last failing check? Maybe I need to fix `This branch is out-of-date with the base branch` by `Merge the latest changes from master into this branch.`, but I'm not sure.

**Comment by sw005320** [comment]
You don't need it.
I can merge this PR.
Thanks, @kkim-asapp

**Comment by sw005320** [comment]
Please make a follow-up PR to provide an updated README file etc.
